sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XSDToSchema.scala

```diff
@@ -35,34 +38,32 @@ import org.apache.spark.sql.types._
 object XSDToSchema {
 
   /**
-   * Reads a schema from an XSD file.
+   * Reads a schema from an XSD path.
    * Note that if the schema consists of one complex parent type which you want to use as
    * the row tag schema, then you will need to extract the schema of the single resulting
    * struct in the resulting StructType, and use its StructType as your schema.
    *
-   * @param xsdFile XSD file
+   * @param xsdPath XSD path
    * @return Spark-compatible schema
    */
-  def read(xsdFile: File): StructType = {
+  def read(xsdPath: Path): StructType = {
+    val in = try {
+      // Handle case where file exists as specified
+      val fs = xsdPath.getFileSystem(SparkHadoopUtil.get.conf)
+      fs.open(xsdPath)
+    } catch {
+      case _: Throwable =>
```

Comment:
I think you should probably rethrow the exception if that's not found in `SparkFiles`?

Reply:
e.g.,

```scala
    } catch {
      case e: Throwable =>
        logInfo(s"$xsdPath was not found, falling back to look up files added by Spark")
        val f = new File(SparkFiles.get(xsdPath.toString))
        if (f.exists()( {
          new FileInputStream(f)
        } else {
          throw e
        }
```

Not tested :-).